test/groovy/BuildExecuteTest.groovy

```diff
@@ -107,6 +110,46 @@ class BuildExecuteTest extends BasePiperTest {
         assertThat(buildToolCalled, is(true))
     }
 
+    @Test
+    void testNpmWithScripts() {
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], {m ->
+            assertEquals('foo', m['runScripts'][0])
+            assertEquals('bar', m['runScripts'][1])
+            return
+        })
+        stepRule.step.buildExecute(
+            script: nullScript,
+            buildTool: 'npm',
+            npmRunScripts: ['foo', 'bar']
+        )
+    }
+
+    @Test
+    void testNpmWithInstallFalse() {
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], {m ->
+            assertFalse(m['install'])
+            return
+        })
+        stepRule.step.buildExecute(
+            script: nullScript,
+            buildTool: 'npm',
+            npmInstall: false
+        )
+    }
+
+    @Test
+    void testNpmWithInstallTrue() {
+        helper.registerAllowedMethod('npmExecuteScripts', [Map.class], {m ->
+            assertTrue(m['install'])
+            return
+        })
+        stepRule.step.buildExecute(
+            script: nullScript,
+            buildTool: 'npm',
+            npmInstall: true
+        )
+    }
```

Comment:
If there is something wrong with the method registration (seen it happing more than once) then the code with the asserts will not be executed. I would re-phrase the test so that you can assert outside the registered closure. Something like:

```
boolean installSet = true
helper.registerAllowedMethod('npmExecuteScripts', [Map.class], {m ->
    installSet = m['install']
})
...
assertFalse(installSet)
```